dali/base/dautils.cpp

```diff
@@ -1991,8 +1992,8 @@ IRemoteConnection *getElementsPaged( IElementsPager *elementsPager,
     if (checkConn && !elem->conn)
         return NULL;
     unsigned n;
-    if (total)
-        *total = elem->totalres.ordinality();
+    if (returned)
+        *returned = elem->totalres.ordinality();
```

Comment:
ok, yes it will be if paging..
But I'm not sure what the point of it is...

We now have:
1) results.ordinality()    ( <=pageSize)
2) total  (a count of _all_ matching files from)
3) returned  ( <= total , could be > pageSize)

Previous you had a total only (and it was uncapped) and adjusted by the postfilter.
That's all you still want..
And you now have it via the getTotalElements() (see other comments re. other implementations getTotalElements()) - but the result returned here should still be adjusted by the postfilter as it was before.